### PR TITLE
Enable using existing namespace for must-gather operations

### PR DESCRIPTION
**Pull request by Ajpantuso**
### Summary

One of the OSD SRE teams (MT-SRE) has a requirement that custom `must-gather` images must be run for troubleshooting managed services on OSD. The `must-gather` scripts run against namespaces in which we are admins, but the `oc adm must-gather` command requires dynamic generation of temporary namespaces/clusterRoleBindings to run `must-gather` pods. This is not possible with _namespace scoped_ admin access so this PR seeks to add an optional `namespace` parameter to the `oc adm must-gather` command so that users can specify an existing namespace for `must-gather` pods to run.

### CODE REVIEW

File: pkg/cli/admin/mustgather/mustgather.go [hunk 1]
```diff
@@ -95,6 +95,7 @@ func NewMustGatherCommand(f kcmdutil.Factory, streams genericclioptions.IOStream
 	cmd.Flags().StringVar(&o.DestDir, "dest-dir", o.DestDir, "Set a specific directory on the local machine to write gathered data to.")
 	cmd.Flags().StringVar(&o.SourceDir, "source-dir", o.SourceDir, "Set the specific directory on the pod copy the gathered data from.")
 	cmd.Flags().StringVar(&o.timeoutStr, "timeout", "10m", "The length of time to gather data, like 5s, 2m, or 3h, higher than zero. Defaults to 10 minutes.")
+	cmd.Flags().StringVar(&o.RunNamespace, "run-namespace", o.RunNamespace, "An existing namespace where must-gather pods should run. If not specified a temporary namespace will be generated.")
```
[the remark this answers is not in the document]
Comment (Ajpantuso) [reply pkg/cli/admin/mustgather/mustgather.go:98]: Both recommendations are now implemented.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**